packages/alfa-dom/src/jsx.ts

```diff
@@ -33,7 +35,11 @@ export function jsx<N extends string = string>(
         continue;
 
       case "box":
-        box = Rectangle.from(value as Rectangle.JSON);
+        const deviceAndBox = value as { device: Device } & Rectangle.JSON;
+
+        box = Rectangle.from(deviceAndBox);
+        device = deviceAndBox.device;
```

Comment:
**Question:** This seems needlessly complex to add the device to `deviceAndBox` and extract it directly.
Isn't it sufficient to just pass it when calling `h`?

Reply:
I am not sure I completely get what you mean by that.

Is it the fact that `device` is put inside the `box` attribute like so `<div box={{ device, x: 8, ... }}>` and would it be better to have it as a separate attribute like so `<div device={{ device }} box={{ x: 8, ... }}>`? 

In any case we need to extract the device somehow inside the `jsx` function so that it can be passed on to `h` as far as I can tell.

It would of course be nice to only have to pass the device in one time like so `h.document(<div box={{...}}></div>, device)`, but the problem is that the elements have already been constructed by `jsx` before `h.document` is called and it's too late to pass in the device because the boxes does not get stored on the elements if there is no device at construction time 🤔